java/core/src/java/org/apache/orc/impl/ParserUtils.java

```diff
@@ -459,10 +459,31 @@ public static ColumnVector[] findColumnVectors(TypeDescription schema,
                                                  boolean isCaseSensitive,
                                                  VectorizedRowBatch batch) {
     List<String> names = ParserUtils.splitName(source);
-    ColumnFinder result = new ColumnFinder(schema, batch, names.size());
+    ColumnFinder result = new ColumnFinder(schema, removeAcidColumnsFromVectorizedRowBatch(schema, batch), names.size());
     findColumn(removeAcid(schema), names, isCaseSensitive, result);
     return result.result;
   }
+  
```

Comment:
I would think we should go a different route. My proposal would be something like.

If the schema is acid schema then when you are looking up column names prefix them with `row.` and just use the full schema. This will ensure that we respect the nesting of the vectors under the struct column vector related to `row`

Reply:
Changed according to this comment.